test/sdk-benchmarks/src/main/java/software/amazon/awssdk/benchmark/BenchmarkRunner.java

```diff
@@ -0,0 +1,114 @@
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.benchmark;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import org.openjdk.jmh.results.RunResult;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.ChainedOptionsBuilder;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import software.amazon.awssdk.utils.Logger;
+
+public class BenchmarkRunner {
+
+    private static final String BENCHMARK_ARG = "-benchmarks";
+    private static final Logger log = Logger.loggerFor(BenchmarkRunner.class);
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    private final List<String> benchmarksToRun;
+
+    public BenchmarkRunner(List<String> benchmarksToRun) {
+        this.benchmarksToRun = benchmarksToRun;
+    }
+
+    public static void main(String... args) throws RunnerException, IOException {
+        if (args.length == 0) {
+            log.error(() -> "No Argument provided");
+            System.exit(0);
+        }
+
+        List<String> benchmarksToRun = parseBenchmarkArgs(args);
+        BenchmarkRunner runner = new BenchmarkRunner(benchmarksToRun);
+        runner.runBenchmark();
+    }
+
+    private void runBenchmark() throws IOException, RunnerException {
+        ChainedOptionsBuilder optionsBuilder = new OptionsBuilder();
+        List<BenchmarkScore> benchmarkScores = retrieveBenchmarkBaseline();
+
+        benchmarksToRun.forEach(optionsBuilder::include);
+
+        log.info(() -> "benchmarks to run " + benchmarksToRun);
+
+        Collection<RunResult> results = new Runner(optionsBuilder.build()).run();
+
+        BenchmarkResultProcessor processor = new BenchmarkResultProcessor(benchmarkScores);
+
+        List<String> failedBenchmarkResults = processor.processBenchmarkResult(results);
+
+        if (!failedBenchmarkResults.isEmpty()) {
+            throw new RuntimeException("Failed the benchmark regression " + failedBenchmarkResults);
+        }
+
+        log.info(() -> "Passed the performance regression tests!");
+    }
+
+    private List<BenchmarkScore> retrieveBenchmarkBaseline() throws IOException {
+        List<BenchmarkScore> benchmarkScores;
+
+        try {
+            benchmarkScores = MAPPER.readValue(getClass().getClassLoader().getResourceAsStream("baseline_score.json"),
+                                               new TypeReference<List<BenchmarkScore>>() {});
+
+        } catch (Exception ex) {
+            log.error(() -> "Failed to retrieve the baseline score", ex);
+            throw ex;
+        }
+
+        return benchmarkScores;
+    }
+
+    /**
+     * Parsing the args. eg: -Dbenchmarks=ApiCallSyncHttpClientBenchmark, ClientCreationBenchmark
```

Comment:
nit: this doesn't seem consistent with actual value of `BENCHMARK_ARG` which is `-benchmark`